src/EditorFeatures/CSharpTest/Classification/SemanticClassifierTests.cs

```diff
@@ -3612,5 +3612,38 @@ void N()
                 TypeParameter("T"),
                 Keyword("notnull"));
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Classification)]
+        public async Task DiscardInAssignment()
+        {
+            await TestAsync(@"
+class X
+{
+    void N()
+    {
+        int.TryParse("""", out var _);
+    }
+}",
+            Method("TryParse"),
+            Static("TryParse"),
+            Keyword("var"),
+            Keyword("_"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Classification)]
+        public async Task ShortDiscardInAssignment()
+        {
+            await TestAsync(@"
+class X
+{
+    void N()
+    {
+        int.TryParse("""", out _);
+    }
+}",
+            Method("TryParse"),
+            Static("TryParse"),
+            Keyword("_"));
+        }
     }
 }
```

Comment:
Here are some more scenarios to test (different contexts where discards are allowed):
- out variable declarations, such as `bool found = TryGetValue(out var _)` or `bool found = TryGetValue(out _)`
- deconstruction assignments, such as `(x, _) = deconstructable;`
- deconstruction declarations, such as `(var x, var _) = deconstructable;`
- is patterns, such as `x is int _`
- switch/case patterns, such as `case int _:`
- lambda parameters, such as `(_, _) => body;` or `(int _, int _) => body;`

Reply:
@jcouv Was tehre any work done to make `_` in a lambda a discard?

Reply:
There was a little: https://github.com/dotnet/roslyn/pull/38786

Reply:
First of all, thank you for providing a list of test cases.

> lambda parameters, such as (_, _) => body; or (int _, int _) => body;
Is this actually a discarding case? for me this `Func<int, int, int> a = (int _, int __) => { return _ + __; };` compiles and I can use the "discard" in the body. I also get a CS0100 if i define it whith two single underscores.

Reply:
This is not a discarding case: `Func<int, int, int> a = (int _, int __) => { return _ + __; };`
But this is: `Func<int, int, int> a = (int _, int _) => { return /* cannot reference either discards here */; };`. This requires language version Preview (otherwise it's a compilation error).

Reply:
FWIW, to recognize discard parameters in lambdas, the key API is `IParameterSymbol.IsDiscard`.